Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -1447,7 +1447,7 @@ template <class TreeBuilder> TreeStatement Parser<LexerType>::parseForStatement(
             return lexicalVariables;
         }
         return { };
-    };
```

Comment:
I think we should keep this semicolon because it is an assignment.
See https://webkit.org/code-style-guidelines/#other-punctuation, which has "auto a = [] { };"